### ISSUE TITLE
Support in-container TLS termination with ACM integration

### ISSUE THREAD

**Issue by thedevopsmachine** [aws/containers-roadmap#1998]
I have a compliance requirement to implement end-to-end TLS, which means I cannot terminate TLS at the load balancer, I need to do so in a container. I would like to be able to use public ACM-issued certificates in these containers, similar to how the Nitro Enclave allows you to use ACM certificates on EC2 instances. Ideally, I'd like to be able to use a Nitro Enclave, but any solution that allows me to terminate TLS in a Fargate container using ACM public certificates is fine.

**Comment by herrhound** [comment]
Hi @thedevopsmachine, do you specifically need to terminate public certs? I'm curious about your use case, and I wonder if using private certs from AWS PCA would address it.

**Comment by thedevopsmachine** [comment]
Yes, the requirement is public certificates. I'm from an internal team so I can't talk about the use case here, but I can make up a realistic one:

TLS termination on ELB's don't support FIPS (a compliance requirement for FedRAMP and some other compliance frameworks). If FIPS compliance is required by a customer, and they are running on Fargate, TLS must be terminated in a container. Customers have no choice but to use a third party CA, since ACM doesn't support this use case. More importantly, they have to use a system such as Secrets Manager to store their private keys, which can also be retrieved (and exported) by humans who have permissions to retrieve the secret (or alter IAM policies). ACM's Enclave mechanism only allows access to the keys from the Enclave, so no matter what permissions the human has, they can't get the private key. It's a far superior mechanism for private key storage.

The ability to export certificates from ACM would work, but it's not a complete solution, because customers will still need to rotate certificates (which will probably require a re-deployment of the service). The Nitro Enclave mechanism automatically takes care of this burden, so customers can do a one-time setup and then forget about it.
